src/assets/scripts/client/trafficGenerator/SpawnScheduler.js

```diff
@@ -12,49 +12,70 @@ import { FLIGHT_CATEGORY } from '../constants/aircraftConstants';
  *
  * @class SpawnScheduler
  */
-export default class SpawnScheduler {
+class SpawnScheduler {
     /**
      * @constructor
      * @for SpawnScheduler
-     * @param spawnPatternCollection {SpawnPatternCollection}
      * @param aircraftController {AircraftController}
      */
-    constructor(spawnPatternCollection, aircraftController) {
-        if (!(spawnPatternCollection instanceof SpawnPatternCollection)) {
-            throw new TypeError('Invalid parameter. SpawnScheduler requires an instance of a SpawnPatternCollection.');
-        }
+    constructor() {
+        /**
+         * @property _aircraftController
+         * @type {AircraftController}
+         * @default null
+         * @private
+         */
+        this._aircraftController = null;
```

Comment:
Not sure if I get your point here. The reference to `AircraftController` has been there before. All I did, was converting it from a method parameter to a class member. `SpawnScheduler` is reinitiated when the airport changes and the reference to `AircraftController` is replaced with the new one.

I don't think it's going to be easy to come up with an `EventBus`-based soluation but I am always open for suggestions 😉

Reply:
You are correct, previously there was a reference to the `AircraftController`, but it was a temporary reference used only during instantiation and never stored.  That doesn't quite work any more, for obvious reasons.  My concern, and what I was getting at before, is that this class now also maintains a reference to the `AircraftController`.  That by itself isn't a huge deal.  The big deal, I feel, is how much it bloats this class by keeping that reference around.  Storing the reference makes this object *much* bigger than it needs to be be cause the `AircraftController` is a large class.  It includes *every* aircraft reference and storing it here means a lot more data.

Log out the `SpawnScheduler` and see just how massive that object is with a reference to the `AircraftController`

Reply:
Isn't it still just a pointer to the aircraftcontroller? Meaning it may be counting it as part of its "size" but it's not _actually_ contributing anything to the objects size, nor storing an actual copy of the aircraftcontroller or anything like that?

Reply:
I would prefer to do this in a new issue if it really becomes an issue.